### PR TITLE
feat(container): support seccompProfile in container security context

### PR DESCRIPTION
**Pull request by serhatcetinkaya**
this adds support for SeccompProfile in container security context.

please refer to:
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.30/#seccompprofile-v1-core

this should also be backported to 1.29 and 1.28 main branches

### CODE REVIEW

File: test/container.test.ts [hunk 1]
```diff
@@ -809,7 +813,34 @@ test('custom security context', () => {
...
+
+  expect(() => new Container({
+    image: 'image',
+    securityContext: {
+      seccompProfile: {
+        type: SeccompProfileType.UNCONFINED,
+        localhostProfile: 'localhostProfile',
+      },
+    },
+  })).toThrowError('localhostProfile must only be set if type is "Localhost"');
```
Comment (iliapolo) [inline test/container.test.ts:846]: Please put this in a separate test.

**Review by iliapolo — CHANGES_REQUESTED** [review]

**Comment by iliapolo** [comment]
Also see https://github.com/cdk8s-team/cdk8s-plus/pull/4500#issuecomment-2283286630. Thanks!

File: test/container.test.ts (hunk 1 above)
Comment (serhatcetinkaya) [reply test/container.test.ts:846]: my mistake, sorry about that. I sent a commit to separate tests

**Comment by serhatcetinkaya** [comment]
@iliapolo thanks for the review, #4500 is merged 👍 

I sent a fix to separate the tests for this one, can you please also take a look at #4483 when you have time? thanks in advance

**Review by iliapolo — APPROVED** [review]

**Comment by iliapolo** [comment]
@serhatcetinkaya this is also need to update from upstream

**Pull request merged**